packages/mcdoc/src/runtime/checker/index.ts

```diff
@@ -564,298 +705,303 @@ function condenseErrorsAndFilterSiblings<T>(
 		// and look sth like this:
 		// Missing either keys ("A", "B" and "C"), ("A", and "D"), or "F"
 		errors.push(
-			...validDefinitions.map(d => ({
-				kind: 'some_missing_keys' as 'some_missing_keys',
-				node: d.node.runtimeNode.node,
-			})),
+			...validDefinitions
+				.flatMap(d =>
+					d.errors
+						.filter(e =>
+							e.kind === 'missing_key' &&
+							!alwaysMissing.some(oe =>
+								oe.node.originalNode === e.node.originalNode
+							)
+						)
+				)
+				.map(e => e.node as RuntimeNode<T>)
+				.concat(
+					validDefinitions.flatMap(d => d.errors).filter(e =>
+						e.kind === 'some_missing_keys'
+					).map(e => e.node as RuntimeNode<T>),
+				)
+				.filter((v, i, arr) =>
+					arr.findIndex(o => o.originalNode === v.originalNode) === i
+				)
+				.map(n => ({
+					kind: 'some_missing_keys' as 'some_missing_keys',
```

Comment:
A good way to prevent some "unsafe" casts is by using `as const` on some of these objects